sessions/forecasting-models.qmd

```diff
@@ -50,690 +56,235 @@ The best way to interact with the material is via the [Visual Editor](https://do
 
-::: {.callout-tip}
 ## Take 5 minutes
-What do you think of the PIT histograms?
-:::
 
-::: {.callout-note collapse="true"}
-## Solution
-What do you think of the PIT histograms?
-
-- The more mechanistic model is reasonably well calibrated but has a slight tendency to be overconfident.
-- The random walk is biased towards overpredicting.
-- The more statistical model is underconfident.
-- Across horizons the more mechanistic model is only liable to underpredict at the longest horizons.
-- The random walk model is initially relatively unbiased and well calibrated but becomes increasingly likely to overpredict as the horizon increases.
-- The forecast date stratified PIT histograms are hard to interpret. We may need to find other ways to visualise bias and calibration at this level of stratification (see the `{scoringutils}` documentation for some ideas).
-
-:::
+None of the forecasts above look great. But which of these models do you think is doing the most reasonable (or least unreasonable) things?
 
-## Scoring on the log scale
-
-Again as in the [forecast evaluation session](forecast-evaluation), we will also score the forecasts on the log scale.
+:::: {.callout-note collapse="true"}
+## Solution
 
-```{r log-convert-forecasts}
-log_sc_forecasts <- sc_forecasts |>
-  transform_forecasts(
-    fun = log_shift,
-    offset = 1,
-    append = FALSE
-  )
+Once again, there is not exactly a "right" answer here, but here are a few things to consider:
 
-log_sc_scores <- log_sc_forecasts |>
-  score()
-```
+ - The fact that both of the `_trans` forecasts (the ones with the fourth-root transformation) stay positive seems pretty critical. The other forecasts routinely predict impossible values!
+ - The `arima210_trans` model has quite high uncertainty at longer horizons in a way that seems reasonable.
+ 
+Again, all of these forecasts are not very good, but they are progress.
+::::
 
+# Doing something about seasonality
 
-::: {.callout-tip}
-Reminder: For more on scoring on the log scale see @bosse2023scorin on scoring forecasts on transformed scales.
-:::
+The seasonality feature is a critical part of these data. 
+Let's do something to model it!
 
-### At a glance
-
-```{r}
-log_sc_scores |>
-  summarise_scores(by = "model")
-```
-
-::: {.callout-tip}
-## Take 2 minutes
-Before we look in detail at the scores, what do you think the scores are telling you? Which model do you think is best?
-:::
+There is a whole part of ARIMA modeling that accounts for seasonality (they are called Seasonal ARIMA or SARIMA models). 
+However, using seasonal ARIMA models with weekly data is complicated, because some years have 52 weeks and some have 53.
+Here are two relatively straight-forward ways that you could address this:
 
-### CRPS
+1. Define a "season-year" where "week 1" is the start of each season (maybe the first week of September) and count to 52 from there. If there is a 53rd week, drop it. Model with a periodicity of 52.
```

Comment:
I thought the plan was to pyut this in the intro and eval of more complex models?

Reply:
well, we aren't actually doing item 1 here, but yeah, I made the executive decision to include some seasonality in this session so we could have something resembling a "reasonable" model by the end of session 2.